src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -230,6 +255,10 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     protected Query doToQuery(QueryShardContext context) {
         MappedFieldType mappedFieldType = context.fieldMapper(this.fieldName);
 
+        if (mappedFieldType == null && ignoreUnmapped) {
```

Comment:
what was the behavior for case when `mappedFieldType == null` before this change? maybe add a comment that describes such behavior

Reply:
It would throw the exception below on line 263.